src/Engine/ProtoCore/Reflection/MirrorDataInProgress.cs

```diff
@@ -0,0 +1,27 @@
+﻿using ProtoCore.DSASM;
+using ProtoCore.Mirror;
+
+namespace ProtoCore
+{
+    /// <summary>
+    /// WARNING: this class is used as temp value for CachedValue in NodeModel.
+    /// Don't try to get some value, in any case it will be something wrong.
+    /// </summary>
+    public class MirrorDataInProgress : MirrorData
```

Comment:
Can we move this class to somewhere else? Engine is a separate component, it shouldn't have much knowledge about Dynamo